applications/FSIApplication/python_scripts/trilinos_partitioned_fsi_dirichlet_neumann_solver.py

```diff
@@ -17,7 +17,26 @@ def CreateSolver(model, project_parameters):
 class TrilinosPartitionedFSIDirichletNeumannSolver(trilinos_partitioned_fsi_base_solver.TrilinosPartitionedFSIBaseSolver):
     def __init__(self, model, project_parameters):
         super(TrilinosPartitionedFSIDirichletNeumannSolver, self).__init__(model, project_parameters)
-        self._PrintInfoOnRankZero("::[TrilinosPartitionedFSIDirichletNeumannSolver]::", "Solver construction finished.")
+        KratosMultiphysics.Logger.PrintInfo("::[TrilinosPartitionedFSIDirichletNeumannSolver]::", "Solver construction finished.")
+
+    @classmethod
+    def GetDefaultSettings(cls):
+        """This function returns the default-settings used by this class
+        """
+        return KratosMultiphysics.Parameters("""{
+            "echo_level": 0,
+            "parallel_type": "MPI",
```

Comment:
same, then I guess you wouldn't need to do this twice

BTW why does this not call to the baseclass? Like the other solvers e.g. in structural do?

Reply:
I need to do it twice because the parallel type is different in the Trilinos and serial version of the solver.

The key is that I need to have the parallel type in the FSI solver level (instead of only having it in the stage one as usual) to construct the fluid, structure and mesh subdomain solvers by calling the `CreateSolverByParameters` of the wrappers.

Reply:
ok fine, but then I would do the following:

baseclass (`PartitionedFSIDirichletNeumannSolver`):
~~~py
@classmethod
def GetDefaultSettings(cls):
    """This function returns the default-settings used by this class
    """
    this_defaults =  KratosMultiphysics.Parameters("""{
        "echo_level": 0,
        "parallel_type": "OpenMP",
        "solver_type": "partitioned",
        "coupling_scheme": "dirichlet_neumann",
        "structure_solver_settings": {
        },
        "fluid_solver_settings":{
        },
        "mesh_solver_settings":{
        },
        "coupling_settings":{
        }
    }""")
    this_defaults.AddMissingParameters(super(PartitionedFSIDirichletNeumannSolver, cls)._GetDefaultSettings())
    return this_defaults
~~~


derived class (`TrilinosPartitionedFSIDirichletNeumannSolver`):
~~~py
@classmethod
def _GetDefaultSettings(cls):
    this_defaults = KratosMultiphysics.Parameters("""{
        "parallel_type": "MPI",
    }""")
    this_defaults.AddMissingParameters(super(TrilinosPartitionedFSIDirichletNeumannSolver, cls)._GetDefaultSettings())
    return this_defaults
~~~

this way you don't have to duplicate anything

Reply:
Cool. Let's do it in that way then.